### PR TITLE
testing/e2e,grpcproxy: Fix: go test --tags "cluster_proxy" -v ./tests/e2e/...

### PR DESCRIPTION
**Pull request by ptabor**
PR contains fixes that makes the e2e tests work through grpc proxy: 

In particular: 
  - Connection between proxy and etcd-server is using a proper CN-less certificate. 
  - Spawned etcd-proxy service expects proper line: return v2p.waitReady("httpproxy: endpoints found")

Among others addresses requests: 
  - https://github.com/etcd-io/etcd/issues/11970: running grpc proxy that **exposes** CN cert, but no CN-cert to speak to etcd-server
  - make sure modularization (https://github.com/etcd-io/etcd/pull/12204) is not introducing any regressions and all the tests pass (even if they didn't prior to the refactoring).
  
Tested with: 
  go test --tags "cluster_proxy" -v ./tests/e2e/...

### CODE REVIEW

**Comment by ptabor** [comment]
@mitake @jingyih @jpbetz - please take a look.

**Comment by jpbetz** [comment]
@wenjiaswe Would you be willing to look at the certs related changes?

**Review by jpbetz — CHANGES_REQUESTED** [review]
One very minor question about a comment then LGTM

File: tests/e2e/cluster_proxy_test.go [hunk 1]
```diff
@@ -204,7 +204,8 @@ func (v2p *proxyV2Proc) Start() error {
 	if err := v2p.start(); err != nil {
 		return err
 	}
-	return v2p.waitReady("httpproxy: endpoints found")
+	// "caller":"httpproxy/director.go:65","msg":"endpoints found","endpoints":["http://localhost:20000"]}
```
Comment (jpbetz) [inline tests/e2e/cluster_proxy_test.go:207]: Remove?

**Review by jpbetz — CHANGES_REQUESTED** [review]

File: tests/e2e/cluster_proxy_test.go (hunk 1 above)
Comment (ptabor) [reply tests/e2e/cluster_proxy_test.go:208]: Done. PTAL.

**Pull request merged**